### PR TITLE
Fix bug when running setup.py without numpy installed

### LINKED ISSUE #402: Running setup.py in affiliated package is not possible unless numpy is already installed
In a clean environment, running `setup.py` in any affiliated package will cause something like the error listed below. This is obviously unintuitive and undesirable behavior. 

**EDIT**: This apparently only affects `setup.py install/develop`, but not `setup.py egg_info`, which is why it has not been detected in Astropy's CI builds.

It's not yet clear to me yet whether this can be fixed by putting some `try/except` logic around the imports below (and possibly elsewhere), or whether this is a more fundamental problem with the helpers strategy: https://github.com/astropy/astropy-helpers/blob/e2b5151a33f1aafa0a19134c76d0ec9eb9d75221/astropy_helpers/utils.py#L59-L79

```
running build_ext
Traceback (most recent call last):
  File "./setup.py", line 143, in <module>
    **package_info
  File "/Users/ddavella/miniconda3/envs/ah-bottleneck/lib/python3.6/site-packages/setuptools/__init__.py", line 140, in setup
    return distutils.core.setup(**attrs)
  File "/Users/ddavella/miniconda3/envs/ah-bottleneck/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/Users/ddavella/miniconda3/envs/ah-bottleneck/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/Users/ddavella/miniconda3/envs/ah-bottleneck/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/ddavella/miniconda3/envs/ah-bottleneck/lib/python3.6/site-packages/setuptools/command/develop.py", line 38, in run
    self.install_for_development()
  File "/Users/ddavella/miniconda3/envs/ah-bottleneck/lib/python3.6/site-packages/setuptools/command/develop.py", line 138, in install_for_development
    self.run_command('build_ext')
  File "/Users/ddavella/miniconda3/envs/ah-bottleneck/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/ddavella/miniconda3/envs/ah-bottleneck/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/ddavella/sandbox/ah-bottleneck/astropy_helpers/astropy_helpers/commands/build_ext.py", line 292, in run
    np_include = get_numpy_include_path()
  File "/Users/ddavella/sandbox/ah-bottleneck/astropy_helpers/astropy_helpers/utils.py", line 72, in get_numpy_include_path
    import numpy
ModuleNotFoundError: No module named 'numpy'
```

### PR DESCRIPTION
**Pull request by drdavella**
This fixes #402. Thanks @saimn for identifying the fix.

### CODE REVIEW

**Comment by bsipocz** [comment]
Could we test this?

**Comment by drdavella** [comment]
@bsipocz I think to test this we might need to add a job to the CI matrix that doesn't install `numpy` with `conda`?

**Comment by drdavella** [comment]
Can confirm that running the tests on `master` in an env without `numpy` fails, but with this branch passes, so I updated the CI matrix.

File: .travis.yml [hunk 1]
```diff
@@ -37,6 +37,10 @@ matrix:
     - os: linux
       env: PYTHON_VERSION=3.6 PIP_DEPENDENCIES='git+https://github.com/sphinx-doc/sphinx.git#egg=sphinx coveralls'
            CONDA_DEPENDENCIES="setuptools cython numpy pytest-cov"
+    # Test without installing numpy beforehand to make sure everything works
+    # without assuming numpy is already installed
+    - os: linux
+      env: PYTHON_VERSION=3.6 CONDA_DEPENDENCIES='sphinx cython pytest-cov'
```
Comment (pllim) [inline .travis.yml]: The the additional `CONDA_DEPENDENCIES` necessary?

Comment (drdavella) [reply .travis.yml]: The point is to leave out `numpy` for this test case.

**Comment by bsipocz** [comment]
Thank you @drdavella!

**Pull request merged**